TimeAlarms mock missing AlarmID_t, dtINVALID_ALARM_ID, enable() and disable()

## Description

The `TimeAlarms.h` mock in ArduinoMock is currently a stub — methods exist but do nothing functional. This means alarm-driven code cannot be tested or simulated natively at all.

We need a **real working mock** that actually fires callbacks on a native timer, so simulator and test environments behave like real hardware.

## Current mock (stub only)

```cpp
class AlarmClass {
 public:
  void delay(unsigned long ms) { ::delay(ms); }
  int timerRepeat(unsigned long seconds, OnTick_t callback) {
    (void)seconds; (void)callback;
    return _nextId++;  // callback is never stored or called
  }
  int timerOnce(unsigned long seconds, OnTick_t callback) {
    (void)seconds; (void)callback;
    return _nextId++;  // callback is never stored or called
  }
  bool free(int id) { (void)id; return true; }
  void reset() { _nextId = 0; }
};
```

## What we need

A native implementation that:

1. **Stores registered callbacks** with their intervals
2. **Fires callbacks** when their interval elapses (using `std::chrono` or similar)
3. **Supports `enable(id)` / `disable(id)`** to pause and resume individual alarms without losing their registration
4. **`Alarm.delay(ms)`** should tick the alarm scheduler while waiting, just like the real library

### Minimum API surface to match (from real TimeAlarms)

```cpp
typedef uint8_t AlarmID_t;
constexpr AlarmID_t dtINVALID_ALARM_ID = 255;

class AlarmClass {
 public:
  AlarmID_t timerRepeat(time_t seconds, OnTick_t callback);  // fires every N seconds
  AlarmID_t timerOnce(time_t seconds, OnTick_t callback);    // fires once after N seconds
  void enable(AlarmID_t id);
  void disable(AlarmID_t id);
  bool free(AlarmID_t id);
  void delay(unsigned long ms);  // ticks the scheduler while waiting
};
```

### Suggested native implementation approach

```cpp
#include <chrono>
#include <thread>

struct AlarmEntry {
  OnTick_t callback;
  unsigned long intervalMs;
  std::chrono::steady_clock::time_point nextFire;
  bool enabled;
  bool repeat;
};

class AlarmClass {
  std::array<AlarmEntry, 8> _alarms{};
  int _nextId = 0;

  void tick() {
    auto now = std::chrono::steady_clock::now();
    for (auto& a : _alarms) {
      if (!a.enabled || !a.callback) continue;
      if (now >= a.nextFire) {
        a.callback();
        if (a.repeat) a.nextFire = now + std::chrono::milliseconds(a.intervalMs);
        else a.enabled = false;
      }
    }
  }

 public:
  AlarmID_t timerRepeat(time_t seconds, OnTick_t callback) {
    AlarmID_t id = _nextId++;
    _alarms[id] = { callback, (unsigned long)seconds * 1000,
                    std::chrono::steady_clock::now() + std::chrono::seconds(seconds),
                    true, true };
    return id;
  }

  AlarmID_t timerOnce(time_t seconds, OnTick_t callback) {
    AlarmID_t id = _nextId++;
    _alarms[id] = { callback, 0,
                    std::chrono::steady_clock::now() + std::chrono::seconds(seconds),
                    true, false };
    return id;
  }

  void enable(AlarmID_t id)  { _alarms[id].enabled = true; }
  void disable(AlarmID_t id) { _alarms[id].enabled = false; }
  bool free(AlarmID_t id)    { _alarms[id] = {}; return true; }

  void delay(unsigned long ms) {
    auto end = std::chrono::steady_clock::now() + std::chrono::milliseconds(ms);
    while (std::chrono::steady_clock::now() < end) {
      tick();
      std::this_thread::sleep_for(std::chrono::milliseconds(1));
    }
  }
};
```

## Errors currently produced without this fix

```
src/routines.h:37:1: error: unknown type name 'AlarmID_t'
src/routines.h:37:39: error: use of undeclared identifier 'dtINVALID_ALARM_ID'
src/routines.h:52:31: error: no member named 'enable' in 'AlarmClass'
src/routines.h:64:9: error: no member named 'disable' in 'AlarmClass'
```

## Context

Encountered in a simulator environment (`env:simulator`) using ArduinoMock at `rc-0.2.0`. Alarm-driven periodic broadcasting (GPS position send every 3s) needs to actually fire natively for the simulator to be useful.

--- a/src/TimeAlarms.h
+++ b/src/TimeAlarms.h
@@ -1,32 +1,90 @@
 #pragma once
+#include <array>
+#include <chrono>
 #include <cstdint>
 #include <functional>
+#include <thread>
 
 #include "times.h"
 
-typedef void (*OnTick_t)();
+typedef std::function<void()> OnTick_t;
+
+typedef uint8_t AlarmID_t;
+constexpr AlarmID_t dtINVALID_ALARM_ID = 255;
+
+static constexpr int MAX_ALARMS = 8;
+
+struct AlarmEntry {
+  OnTick_t callback;
+  unsigned long intervalMs = 0;
+  std::chrono::steady_clock::time_point nextFire;
+  bool enabled = false;
+  bool repeat = false;
+};
 
 class AlarmClass {
  public:
-  void delay(unsigned long ms) { ::delay(ms); }
-  int timerRepeat(unsigned long seconds, OnTick_t callback) {
-    (void)seconds;
-    (void)callback;
-    return _nextId++;
+  AlarmID_t timerRepeat(time_t seconds, OnTick_t callback) {
+    if (_nextId >= MAX_ALARMS) return dtINVALID_ALARM_ID;
+    AlarmID_t id = static_cast<AlarmID_t>(_nextId++);
+    _alarms[id] = {
+        callback, static_cast<unsigned long>(seconds) * 1000,
+        std::chrono::steady_clock::now() + std::chrono::seconds(seconds), true, true};
+    return id;
   }
-  int timerOnce(unsigned long seconds, OnTick_t callback) {
-    (void)seconds;
-    (void)callback;
-    return _nextId++;
+
+  AlarmID_t timerOnce(time_t seconds, OnTick_t callback) {
+    if (_nextId >= MAX_ALARMS) return dtINVALID_ALARM_ID;
+    AlarmID_t id = static_cast<AlarmID_t>(_nextId++);
+    _alarms[id] = {
+        callback, 0, std::chrono::steady_clock::now() + std::chrono::seconds(seconds), true, false};
+    return id;
   }
-  bool free(int id) {
-    (void)id;
+
+  void enable(AlarmID_t id) {
+    if (id < MAX_ALARMS) _alarms[id].enabled = true;
+  }
+
+  void disable(AlarmID_t id) {
+    if (id < MAX_ALARMS) _alarms[id].enabled = false;
+  }
+
+  bool free(AlarmID_t id) {
+    if (id >= MAX_ALARMS) return false;
+    _alarms[id] = {};
     return true;
   }
-  void reset() { _nextId = 0; }
+
+  void delay(unsigned long ms) {
+    auto end = std::chrono::steady_clock::now() + std::chrono::milliseconds(ms);
+    while (std::chrono::steady_clock::now() < end) {
+      tick();
+      std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    }
+  }
+
+  void reset() {
+    _nextId = 0;
+    for (auto& a : _alarms) a = {};
+  }
 
  private:
+  std::array<AlarmEntry, MAX_ALARMS> _alarms{};
   int _nextId = 0;
+
+  void tick() {
+    auto now = std::chrono::steady_clock::now();
+    for (auto& a : _alarms) {
+      if (!a.enabled || !a.callback) continue;
+      if (now >= a.nextFire) {
+        a.callback();
+        if (a.repeat)
+          a.nextFire = now + std::chrono::milliseconds(a.intervalMs);
+        else
+          a.enabled = false;
+      }
+    }
+  }
 };
 
 extern AlarmClass Alarm;

--- a/src/TimeAlarms.h
+++ b/src/TimeAlarms.h
@@ -27,17 +27,24 @@ class AlarmClass {
   AlarmID_t timerRepeat(time_t seconds, OnTick_t callback) {
     if (_nextId >= MAX_ALARMS) return dtINVALID_ALARM_ID;
     AlarmID_t id = static_cast<AlarmID_t>(_nextId++);
-    _alarms[id] = {
-        callback, static_cast<unsigned long>(seconds) * 1000,
-        std::chrono::steady_clock::now() + std::chrono::seconds(seconds), true, true};
+    auto& a = _alarms[id];
+    a.callback = callback;
+    a.intervalMs = static_cast<unsigned long>(seconds) * 1000;
+    a.nextFire = std::chrono::steady_clock::now() + std::chrono::seconds(seconds);
+    a.enabled = true;
+    a.repeat = true;
     return id;
   }
 
   AlarmID_t timerOnce(time_t seconds, OnTick_t callback) {
     if (_nextId >= MAX_ALARMS) return dtINVALID_ALARM_ID;
     AlarmID_t id = static_cast<AlarmID_t>(_nextId++);
-    _alarms[id] = {
-        callback, 0, std::chrono::steady_clock::now() + std::chrono::seconds(seconds), true, false};
+    auto& a = _alarms[id];
+    a.callback = callback;
+    a.intervalMs = 0;
+    a.nextFire = std::chrono::steady_clock::now() + std::chrono::seconds(seconds);
+    a.enabled = true;
+    a.repeat = false;
     return id;
   }
 
@@ -51,7 +58,8 @@ class AlarmClass {
 
   bool free(AlarmID_t id) {
     if (id >= MAX_ALARMS) return false;
-    _alarms[id] = {};
+    _alarms[id].callback = nullptr;
+    _alarms[id].enabled = false;
     return true;
   }
 
@@ -65,7 +73,10 @@ class AlarmClass {
 
   void reset() {
     _nextId = 0;
-    for (auto& a : _alarms) a = {};
+    for (auto& a : _alarms) {
+      a.callback = nullptr;
+      a.enabled = false;
+    }
   }
 
  private:

--- a/test/time_alarms_gtest.cpp
+++ b/test/time_alarms_gtest.cpp
@@ -2,6 +2,8 @@
 
 #include "TimeAlarms.h"
 
+// --- existing API ---
+
 TEST(TimeAlarmsTest, TimerRepeatReturnsIncrementingIds) {
   Alarm.reset();
   EXPECT_EQ(Alarm.timerRepeat(10, nullptr), 0);
@@ -18,9 +20,9 @@ TEST(TimeAlarmsTest, TimerOnceReturnsIncrementingIds) {
 
 TEST(TimeAlarmsTest, RepeatAndOnceIdsAreDistinctAndMonotonic) {
   Alarm.reset();
-  int id0 = Alarm.timerRepeat(10, nullptr);
-  int id1 = Alarm.timerOnce(5, nullptr);
-  int id2 = Alarm.timerRepeat(15, nullptr);
+  AlarmID_t id0 = Alarm.timerRepeat(10, nullptr);
+  AlarmID_t id1 = Alarm.timerOnce(5, nullptr);
+  AlarmID_t id2 = Alarm.timerRepeat(15, nullptr);
   EXPECT_EQ(id0, 0);
   EXPECT_EQ(id1, 1);
   EXPECT_EQ(id2, 2);
@@ -30,7 +32,7 @@ TEST(TimeAlarmsTest, RepeatAndOnceIdsAreDistinctAndMonotonic) {
 
 TEST(TimeAlarmsTest, FreeReturnsTrue) {
   Alarm.reset();
-  int id = Alarm.timerRepeat(10, nullptr);
+  AlarmID_t id = Alarm.timerRepeat(10, nullptr);
   EXPECT_TRUE(Alarm.free(id));
 }
 
@@ -50,4 +52,63 @@ TEST(TimeAlarmsTest, ResetResetsIdCounterToZero) {
 TEST(TimeAlarmsTest, GlobalAlarmInstanceIsAccessible) {
   Alarm.reset();
   EXPECT_EQ(Alarm.timerRepeat(1, nullptr), 0);
+}
+
+// --- AlarmID_t and dtINVALID_ALARM_ID ---
+
+TEST(TimeAlarmsTest, AlarmIdTypeExists) {
+  AlarmID_t id = 0;
+  EXPECT_EQ(id, 0);
+}
+
+TEST(TimeAlarmsTest, InvalidAlarmIdConstant) { EXPECT_EQ(dtINVALID_ALARM_ID, 255); }
+
+TEST(TimeAlarmsTest, MaxAlarmsReturnsInvalidId) {
+  Alarm.reset();
+  for (int i = 0; i < MAX_ALARMS; ++i) Alarm.timerRepeat(1, nullptr);
+  EXPECT_EQ(Alarm.timerRepeat(1, nullptr), dtINVALID_ALARM_ID);
+}
+
+// --- enable / disable ---
+
+TEST(TimeAlarmsTest, EnableDisableDoNotThrow) {
+  Alarm.reset();
+  AlarmID_t id = Alarm.timerRepeat(100, nullptr);
+  EXPECT_NO_THROW(Alarm.disable(id));
+  EXPECT_NO_THROW(Alarm.enable(id));
+}
+
+TEST(TimeAlarmsTest, DisabledAlarmDoesNotFire) {
+  Alarm.reset();
+  int count = 0;
+  AlarmID_t id = Alarm.timerRepeat(0, [&]() { count++; });
+  Alarm.disable(id);
+  Alarm.delay(5);
+  EXPECT_EQ(count, 0);
+}
+
+// --- callback firing ---
+
+TEST(TimeAlarmsTest, TimerOnceFiresCallback) {
+  Alarm.reset();
+  int count = 0;
+  Alarm.timerOnce(0, [&]() { count++; });
+  Alarm.delay(10);
+  EXPECT_EQ(count, 1);
+}
+
+TEST(TimeAlarmsTest, TimerRepeatFiresMultipleTimes) {
+  Alarm.reset();
+  int count = 0;
+  Alarm.timerRepeat(0, [&]() { count++; });
+  Alarm.delay(10);
+  EXPECT_GT(count, 1);
+}
+
+TEST(TimeAlarmsTest, TimerOnceFiresOnlyOnce) {
+  Alarm.reset();
+  int count = 0;
+  Alarm.timerOnce(0, [&]() { count++; });
+  Alarm.delay(20);
+  EXPECT_EQ(count, 1);
 }

